snapcraft/internal/project_loader/_config.py

```diff
@@ -260,7 +268,11 @@ def _snapcraft_yaml_load(yaml_file):
 
     try:
         with open(yaml_file, encoding=encoding) as fp:
-            return yaml.load(fp)
+            # Override the loader here so we can parse float as str
+            # which matters when eg. having "version: 2.10"
+            Loader.add_constructor(
+                u'tag:yaml.org,2002:float', float_constructor)
```

Comment:
Doesn't this override the loading of _every_ float to be a string instead? Are we sure that's what we want? Remember that this will affect custom/local plugins as well. If this is the direction we want to go, perhaps this change should be more localized.

Reply:
If this happens, it should be only for `version`; I would rather have a proper schema check, just like we change the on in `version` anyways (allowed/disallowed chars)